src/components/VueGist.vue

```diff
@@ -1,65 +1,109 @@
 <template>
   <div>
-    <app-gist-core :gist-div="gistData" />
+    <template v-if="gistErr">
+      <div class="alert alert-danger">
+        <i class="material-icons">error</i>
+        Oops! We hit a block, Please check the gist-id/filename.
+      </div>
+    </template>
+    <template v-else>
+      <app-gist-core :gist-div="gistData" />
+    </template>
   </div>
 </template>
 
 <script>
 import VueGistCore from "./VueGistCore.vue";
-import jsonp from 'jsonp';
+import jsonp from "jsonp";
 
 export default {
   components: {
-    appGistCore: VueGistCore
+    appGistCore: VueGistCore,
   },
   props: {
     gistId: {
       type: String,
-      required: true
+      required: true,
     },
     file: {
       type: String,
       required: false,
-      default: ""
-    }
+      default: "",
+    },
   },
   data() {
     return {
       gistData: "loading...",
-      gistUrl: "https://gist.github.com/"
+      gistUrl: "https://gist.github.com/",
+      gistErr: false,
     };
   },
   computed: {
     url() {
-      return `${this.gistUrl}${this.gistId}.json`
+      return `${this.gistUrl}${this.gistId}.json`;
     },
     params() {
       if (this.file.length > 0) {
         return `file=${this.file}`;
       }
-      return ''
+      return "";
+    },
+    fullUrl() {
+      return `${this.url}?${this.params}`;
     },
-    fullUrl(){
-      return `${this.url}?${this.params}`
-    }
   },
-    created() {
-    this.getGistData(this.gistId);
+  watch: {
+    gistId: {
+      handler: function () {
+        this.$data.gistErr = false;
+        this.$data.gistData = "loading...";
+        this.getGistData();
+      },
+      immediate: true,
+      deep: true,
+    },
   },
   methods: {
-    getGistData(gistId) {
-      jsonp(this.fullUrl,{ timeout: 20000 }, (err, response) => {
+    getGistData() {
+      jsonp(this.fullUrl, { timeout: 20000 }, (err, response) => {
         if (err) {
-            console.error(err.message);
-            return
+          this.$data.gistErr = true;
+          return;
         }
-          this.gistData = response.div;
-        })
-    }
-  }
+        this.gistData = response.div;
+      });
+    },
+  },
 };
 </script>
 
 <style scoped>
 @import url("https://github.githubassets.com/assets/gist-embed-4ac6018bcc05457cde2f66d2e7299d11.css");
-</style>
\ No newline at end of file
+@import url("https://fonts.googleapis.com/icon?family=Material+Icons");
+
+.alert {
+  min-width: 64px;
+  padding: 12px;
+  margin-bottom: 18px;
+  border: 2px solid transparent;
+  border-radius: 3px;
+}
+.alert-danger {
+  background-color: #e27c79;
+  border-color: #dd6864;
+  color: #801915;
+}
+.alert i {
+  float: left;
+  vertical-align: middle;
+  font-size: 18px;
+}
+.shadow-1, .alert {
+  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
+}
+
+.shadow-2, .alert:hover {
+  box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
+}
+
```

Comment:
Probably avoid importing additional styling. It does look good, but for other users a styling could conflict with their styling.

A better solution could potentially be using the github 404 image.

![image](https://user-images.githubusercontent.com/883233/102043641-d4817580-3d89-11eb-885d-2786373932d4.png)

Again this is just a suggestion.

Reply:
Yeah, This is a good idea, and do you know any API call to get the 404 images from GitHub.

Reply:
Can you please check the pipeline there is an error coming while building danger js and please update the dependencies so many vulnerabilities?